### PR TITLE
[ComboBox] Fix OnValueChanged being called multiple times

### LINKED ISSUE #2851: FluentCombobox OnValueChanged is called twice when the item selected has more than one space in the middle of the items string or an extra space at the beginning or end of the string.
# 🐛 Bug Report

FluentCombobox OnValueChanged is called twice when the item selected has more than one space in the middle of the items string or an extra space at the beginning or end of the string.

When OnValueChanged is called twice, the function is reentered a second time prior to the first call being completed.

This cannot be fixed in user code without elaborate workarounds when data comes from a database or other source.

## 💻 Repro or Code Sample

```
@page "/test"

<FluentCombobox Items="@_items" ValueChanged="@OnValueChanged" />

@code
{
    private List<string> _items = ["good", "good good", "bad  bad", "bad ", " bad"];

    private async Task OnValueChanged(string? value)
    {
        Debug.WriteLine($"Start: '{value}')");

        await Task.Delay(100);

        Debug.WriteLine($"End: '{value}')");
    }
}
```

## 🤔 Expected Behavior

    When you select 'good' you get this expected output in the Output window:
    Start: 'good')
    End: 'good')

    When you select 'good good' you get this expected output in the Output window:
    Start: 'good good')
    End: 'good good')

## 😯 Current Behavior

    When you select 'bad  bad' you get this unexpected output in the Output window:
    Start: 'bad bad')
    Start: 'bad  bad')
    End: 'bad  bad')
    End: 'bad bad')

    When you select 'bad ' you get this unexpected output in the Output window:
    Start: 'bad')
    Start: 'bad ')
    End: 'bad ')
    End: 'bad')

    When you select ' bad' you get this unexpected output in the Output window:
    Start: 'bad')
    Start: ' bad')
    End: 'bad')
    End: ' bad')

## 🌍 Your Environment

.NET 8
Fluent UI Blazor library Version 4.10.2
Blazor Server
Microsoft Edge for Business Version 130.0.2849.52 (Official build) (64-bit)

### PR DESCRIPTION
**Pull request by vnbaaij**
Fix #2851 by not calling `base.ChangeHandlerAsync` when no item selected. This could happen if the  Value and the Text of an option differed.

See original issue for before output. Now all test situations render as expected.

### CODE REVIEW

File: src/Core/Components/List/FluentCombobox.razor.cs [hunk 1]
```diff
@@ -138,16 +138,20 @@ protected override async Task ChangeHandlerAsync(ChangeEventArgs e)
...
             {
                 await OnSelectedItemChangedHandlerAsync(item);
             }
-
         }
     }
 
+    protected override async Task InputHandlerAsync(ChangeEventArgs e)
+    {
+        await base.ChangeHandlerAsync(e);
```
Comment (dvoituron) [inline src/Core/Components/List/FluentCombobox.razor.cs:151]: Why did you remove the usage of `Immediate`?
The method `FluentInputBase.InputHandlerAsync` will not used.

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: src/Core/Components/List/FluentCombobox.razor.cs (hunk 1 above)
Comment (vnbaaij) [reply src/Core/Components/List/FluentCombobox.razor.cs]: Fixed in last commit

**Comment by vnbaaij** [comment]
`Immediate` (and `ImmediateDelay`) now work again for ComboBox. The double calling of `ChangeHadlerAsync` is fixed by using a specific JS custom event handler. The web conponent was returning a incorrect value in the case of leading/trailing and double spaces. We now look at the actual value in the option.

Image below shows output from the test code. At first green arrow, text is being deleted from the text input with `Immediate` set to false. On pressing enter the value is then handled. At second green arrow, `Immediate` is set to true. You then see every typed character appear in the value

image

**Review by dvoituron — APPROVED** [review]
